bundle/jsky.app.ot/src/main/scala/jsky/app/ot/gemini/editor/targetComponent/details2/NonSiderealNameEditor.scala

```diff
@@ -58,12 +58,18 @@ final class NonSiderealNameEditor extends TelescopePosEditor with ReentrancyHack
 
     def updateDesignation(hd: HorizonsDesignation, name: String): HS2[Unit] =
       HS2.delay {
-        val t0 = Target.name.set(spt.getTarget, name)
-        Target.horizonsDesignation.set(t0, Some(hd)).foreach(spt.setTarget)
+        Swing.onEDT {
```

Comment:
It triggers an event that a lot of code seems to assume always comes on the EDT ... most model changes happen due to user input so this is an unusual case. It's not ideal but I think it's probably the safest approach.